src/ast/ast.cpp

```diff
@@ -44,6 +44,10 @@ void FieldAccess::accept(Visitor &v) {
   v.visit(*this);
 }
 
+void ArrayIndex::accept(Visitor &v) {
```

Comment:
nit: I would rename this to `ArrayAccess`, since this class represents the operation and not the index. `ArrayAccess` is also coherent with the existing `FieldAccess` above.

Reply:
I agree, i've made this change. 

I originally thought `Index` was fine as a verb, but I agree that Access is more consistent and less ambiguous.